streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/EagerBufferConfigImpl.java

```diff
@@ -77,4 +87,14 @@ public int hashCode() {
     public String toString() {
         return "EagerBufferConfigImpl{maxRecords=" + maxRecords + ", maxBytes=" + maxBytes + '}';
     }
+
+    @Override
+    public Suppressed.EagerBufferConfig withLoggingDisabled() {
+        return new EagerBufferConfigImpl(maxRecords, maxBytes, topicConfig, false);
```

Comment:
Should we pass in an empty config at this point (and similar points elsewhere)? It seems like it might violate a desired invariant to have logging disabled, but a config present.

Alternatively, you can make this impossible by only storing the config (and making the field nullable, so null means logging is disabled)

Reply:
I implemented the null option because it avoids some duplicate logic.

What worries me is that `BufferConfigInternal.getLogConfig` now might return null. Is there an annotation I should use, should I return an empty map instead, or leave it as-is?

`BufferConfigInternal` now deviates from the `StoreBuilder` interface (may not be a problem) which means that calls to `<some StoreBuilderImpl>.logConfig` return an empty map where  calls to `BufferConfigInternal.getLogConfig` return a null. I'm just trying to avoid breaking convention here :)

Reply:
I agree it's not desirable to return null from `logConfig`. I think returning an empty map should be fine.

Reply:
Unfortunately, we don't want to add extra dependencies to get a usable `@Nullable/@NotNull` annotation, so there's no annotation you could add either way.